tests/tests_integration/test_api/test_data_modeling/test_spaces.py

```diff
@@ -33,6 +33,11 @@ def test_list(self, cognite_client: CogniteClient, cdf_spaces: SpaceList) -> Non
 
         assert _dump(actual_space_in_cdf) == _dump(cdf_spaces)
 
+    def test_list_include_global(self, cognite_client: CogniteClient, cdf_spaces: SpaceList) -> None:
+        spaces = cognite_client.data_modeling.spaces.list(include_global=True)
+
+        assert len(spaces) > len(cdf_spaces)
```

Comment:
These length checks are a recurring theme, and are very brittle. Please be more specific about the assertions. You're also making strong assumptions on the nature of the data produced by the `cdf_spaces` fixture. It's important to design your tests in such a way that they don't risk false successes in the future if there are changes to the data. So one should be extra careful when using widely scoped fixtures.

I can think of several ways this can go wrong. E.g. we decide to filter down or limit the spaces in the `cdf_spaces` (for making the tests independent of other things going on the project for example), then include_global is broken so that it no longer returns global spaces, then this test will still pass.

A better implementation is to explicitly list all spaces with and without `include_global` and then checking that the first collection contains spaces with `is_global=True` and the other doesn't.

Reply:
Good point, a bit quick on writing this a similar tests.